packages/create-sitecore-jss/src/templates/nextjs/scripts/generate-metadata.ts

```diff
@@ -0,0 +1,27 @@
+import fs from 'fs';
+import path from 'path';
+import { Metadata } from '@sitecore-jss/sitecore-jss-nextjs/editing';
+import { getPackagesMetadata } from '@sitecore-jss/sitecore-jss-dev-tools';
+
+/*
+  METADATA GENERATION
+  Generates the /src/temp/metadata.json file which contains application 
+  configuration metadata that is used for Sitecore XM Cloud integration.
+*/
+generateMetadata();
+
+function generateMetadata(): void {
+  const metadata: Metadata = { packages: {} };
+  metadata.packages = getPackagesMetadata();
+  writeMetadata(metadata);
```

Comment:
I would switch our SDK function to be a `getMetadata()` function which returns `Metadata` (shouldn't have any knowledge of "packages" here in the app layer). This way, if we add anything to metadata in the future, there is nothing in the app layer to update.
```suggestion
  const metadata = getMetadata();
  writeMetadata(metadata);
```